BFS_Connected_Components/Q1_number_of_transformations_bfs.py

```diff
@@ -0,0 +1,37 @@
+import collections
+
+# BFS
+def get_min_operations_bfs(start_num, target_num, additive_nums, multiplicative_nums):
+
+    def get_neighbors(num):
+        neighbors = [num * multiplicative_num for multiplicative_num in multiplicative_nums]
+        neighbors.extend([num + additive_num for additive_num in additive_nums])
+        return neighbors
+
+
+    q = collections.deque([(start_num, 0)])
+    seen = set()
+    seen.add(start_num)
+
+    while q:
+        new_num, depth = q.popleft()
+        
+        if new_num == target_num:
+            return depth
+        elif new_num > target_num:
+            continue
+        neighbors = get_neighbors(new_num)
+        
+        for neighbor in neighbors:
+            if neighbor in seen: continue
+            q.append((neighbor, depth + 1))
+            seen.add(neighbor)
+
+    return 0
```

Comment:
Returning 0 implies that `start_num == target_num` is True. Instead, we should return something special to signify that it's impossible to get to target_num from start_num. For example, you can create a constant at the top of this file, such as `CANNOT_REACH_TARGET = -1` and then return this constant.